src/Nethermind/Nethermind.Evm.Test/Tracing/GethLike4byteTracerTests.cs

```diff
@@ -0,0 +1,129 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Collections.Generic;
+using FluentAssertions;
+using Nethermind.Core.Test.Builders;
+using Nethermind.Evm.Precompiles;
+using Nethermind.Evm.Tracing.GethStyle;
+using Nethermind.Evm.Tracing.GethStyle.Custom.Native.Tracers;
+using NUnit.Framework;
+
+namespace Nethermind.Evm.Test.Tracing;
+
+[TestFixture]
+public class GethLike4byteTracerTests : GethLikeNativeTracerTestsBase
+{
+    [Test]
+    public void Trace_call()
+    {
+        byte[] callInput = Prepare.EvmCode
+            .PushData(SampleHexData2)
+            .STOP()
+            .Done;
+
+        byte[] empty6ByteInput = new byte[6];
+
+        byte[] code = Prepare.EvmCode
+            .CallWithInput(TestItem.AddressA, 50000, callInput)
+            .CallWithInput(TestItem.AddressA, 50000, callInput)
+            .CallWithInput(TestItem.AddressA, 50000, empty6ByteInput)
+            .STOP()
+            .Done;
+
+        Dictionary<string, int> expected4ByteIds = new()
+        {
+            { "62b15678-1", 2 },
+            { "00000000-2", 1 }
+        };
```

Comment:
You can also new syntax:

```
        Dictionary<string, int> expected4ByteIds = new Dictionary<string, int>
        {
            ["62b15678-1"] = 2,
            ["00000000-2"] = 1
        };
```

But I think they are similarly readable, so probably ignore it.